src/main/kotlin/com/criptext/mail/scenes/SceneController.kt

```diff
@@ -2,16 +2,16 @@ package com.criptext.mail.scenes
 
 import com.criptext.mail.IHostActivity
 import com.criptext.mail.bgworker.BackgroundWorkManager
-import com.criptext.mail.utils.removedevice.data.RemovedDeviceRequest
-import com.criptext.mail.utils.removedevice.data.RemovedDeviceResult
+import com.criptext.mail.utils.remotechange.data.RemoteChangeRequest
+import com.criptext.mail.utils.remotechange.data.RemoteChangeResult
 
 /**
  * Base class for all the main controllers.
  * Created by sebas on 1/30/18.
  */
 abstract class SceneController {
 
-    var removeDeviceDataSource: BackgroundWorkManager<RemovedDeviceRequest, RemovedDeviceResult>? = null
+    var remoteChangeDataSource: BackgroundWorkManager<RemoteChangeRequest, RemoteChangeResult>? = null
```

Comment:
Why var? and why nullable?

Instead of adding a mandatory data source for each controller, I think it's time to have a single global data source for all. It makes sense since many tasks return results that are still important for the user even if he/she goes to a different activity.

Reply:
Var and nullable because not every scene needs this data source, but the vast majority does, so I thought maybe it's best to leave it like that just for the ones who does need it. I do think the idea of a single data source is good, maybe we can leave this re-factorization for it's own PR? I think that's better than changing this one that is for something else. Because the single global data source is a big one and should be on it's own PR.

Reply:
then it shouldn't be an attribute of `SceneController` just pass it in the controller's constructor like every  other data source.